gslib/tests/testcase/base.py

```diff
@@ -106,8 +107,17 @@ def MakeTempName(self, kind, prefix=''):
       prefix=prefix, method=self.GetTestMethodName(), kind=kind)
     name = name[:MAX_BUCKET_LENGTH-9]
     name = '{name}-{rand}'.format(name=name, rand=self.MakeRandomTestString())
-    # As of March 2018, S3 no longer accepts underscores or uppercase letters in
-    # bucket names.
+    totalNameLen = len(name) + len(suffix)
+    if suffix:
+      if kind == 'bucket' and totalNameLen > MAX_BUCKET_LENGTH:
+        self.fail(
+          'Tried to create a psuedo-random bucket name with a specific '
+          'suffix, but the generated name was too long and there was not '
+          'enough room for the suffix. Please use shorter strings or perform '
+          'name randomization manually.\nRequested name: ' + name)
```

Comment:
Might also be worth including the suffix in the "requested name" part of the error message.

Reply:
Makes sense! Fixed 🙂